### PR TITLE
Export

### PR DESCRIPTION
**Pull request by ghost**
Added export data methods to JSON/XML and XML annotation to all entities.

### CODE REVIEW

File: src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java [hunk 1]
```diff
@@ -11,26 +11,89 @@
...
+        gsonBuilder.registerTypeAdapter(LocalDateTime.class, new LocalDateTimeSerializer());
+        gsonBuilder.registerTypeAdapter(LocalDate.class, new LocalDateDeserializer());
+        gsonBuilder.registerTypeAdapter(LocalDateTime.class, new LocalDateTimeDeserializer());
+        Gson gson = gsonBuilder.setPrettyPrinting().create();
+        myWriter.write(gson.toJson(object));
+        myWriter.close();
+    }
+
+    static class LocalDateSerializer implements JsonSerializer<LocalDate> {
+        private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d-MMM-yyyy", Locale.forLanguageTag("US"));
```
Comment (ghost) [inline src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java:67]: Added "Locale.forLanguageTag("US")" for english month format.

File: src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java [hunk 2]
```diff
@@ -11,26 +11,89 @@
...
+
+        @Override
+        public JsonElement serialize(LocalDate localDate, Type srcType, JsonSerializationContext context) {
+            return new JsonPrimitive(formatter.format(localDate));
+        }
+    }
 
-    ////// EXPORT
+    static class LocalDateTimeSerializer implements JsonSerializer<LocalDateTime> {
+        private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d::MMM::uuuu HH::mm::ss", Locale.forLanguageTag("US"));
```
Comment (ghost) [inline src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java:76]: Same here.

File: src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java [hunk 3]
```diff
@@ -11,26 +11,89 @@
...
+import javax.xml.bind.Marshaller;
+
 public class Export {
 
-    // TODO Move to .env
+    // TODO Move to .env (senior)
     private static final Logger LOGGER = LoggerFactory.getLogger(Export.class);
     public static final String TICKER_PATH = "txt/export/tickers/";
     private static final String STOCK_PATH = "txt/export/stocks/";
     private static final String DELIMITER = "#";
```
Comment (holymiko) [inline src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java:41]: add PDF_PATH = txt/export/pdf/
add JSON_PATH = txt/export/json/
add XML_PATH = txt/export/xml/

File: src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java [hunk 4]
```diff
@@ -11,26 +11,89 @@
...
+    private static final DateTimeFormatter dtf = DateTimeFormatter.ofPattern("ddMMyyyy_HHmmss");
+
+    public void exportToXML(Object object) throws JAXBException {
+        File file = new File(object.getClass().getSimpleName() + "_" + dtf.format(LocalDateTime.now()) +".xml");
```
Comment (holymiko) [inline src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java]: add XML_PATH to name

File: src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java [hunk 5]
```diff
@@ -11,26 +11,89 @@
...
+        JAXBContext jaxbContext = JAXBContext.newInstance(object.getClass());
+        Marshaller jaxbMarshaller = jaxbContext.createMarshaller();
+        jaxbMarshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+        jaxbMarshaller.marshal(object, file);
+    }
+
+    public void exportToJSON(Object object) throws IOException {
+        FileWriter myWriter = new FileWriter(object.getClass().getSimpleName() + "_" + dtf.format(LocalDateTime.now()) +".json");
```
Comment (holymiko) [inline src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java]: add JSON_PATH to name

File: src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java (hunk 3 above)
Comment (ghost) [reply src/main/java/home/holymiko/InvestmentScraperApp/Server/API/TextPort/Export.java:41]: Done

**Pull request closed without merging**